File: efcore-webapp/Data/AppDbContext.vb
Imports EfCoreWebApp.Models
Imports Microsoft.EntityFrameworkCore

Namespace Data
    Public Class AppDbContext
        Inherits DbContext

        Public Sub New(options As DbContextOptions(Of AppDbContext))
            MyBase.New(options)
        End Sub

        Public ReadOnly Property Todos As DbSet(Of Todo)
            Get
                Return Me.Set(Of Todo)()
            End Get
        End Property

        Public ReadOnly Property Products As DbSet(Of Product)
            Get
                Return Me.Set(Of Product)()
            End Get
        End Property
    End Class
End Namespace

File: efcore-webapp/Data/AppDbContext.cs
using EfCoreWebApp.Models;
using Microsoft.EntityFrameworkCore;

namespace EfCoreWebApp.Data;

public class AppDbContext : DbContext
{
    public AppDbContext(DbContextOptions<AppDbContext> options) : base(options) { }
    public DbSet<Todo> Todos => Set<Todo>();
    public DbSet<Product> Products => Set<Product>();
}
